src/Interpreters/InterpreterRenameQuery.cpp

```diff
@@ -127,14 +127,23 @@ BlockIO InterpreterRenameQuery::executeToTables(const ASTRenameQuery & rename, c
         {
             StorageID from_table_id{elem.from_database_name, elem.from_table_name};
             StorageID to_table_id{elem.to_database_name, elem.to_table_name};
-            std::vector<StorageID> ref_dependencies;
-            std::vector<StorageID> loading_dependencies;
+            std::vector<StorageID> from_ref_dependencies;
+            std::vector<StorageID> from_loading_dependencies;
+            std::vector<StorageID> to_ref_dependencies;
+            std::vector<StorageID> to_loading_dependencies;
 
-            if (!exchange_tables)
+            if (exchange_tables)
```

Comment:
Previously we updated dependencies only for `RENAME`, but not for `EXCHANGE` - it was a bug, during `EXCHANGE` we need also to exchange the dependencies